cargo/private/cargo_toml_info/main.rs

```diff
@@ -0,0 +1,276 @@
+        (LintGroup::RustDoc, output_rustdoc_lints),
+        (LintGroup::RustCheckCfg, output_rustc_check_cfg),
+    ];
+
+    let lints = match &crate_manifest.lints {
+        Some(lints) if lints.workspace => {
+            let workspace = workspace_manifest
+                .as_ref()
+                .and_then(|manifest| manifest.workspace.as_ref())
+                .ok_or({
+                    "manifest inherits lints from the workspace, but no workspace manifest provided"
+                })?;
+            workspace.lints.as_ref()
+        }
+        Some(lints) => Some(&lints.groups),
+        None => None,
+    };
+    let Some(lints) = lints else {
+        return Ok(());
+    };
+
+    for (group, path) in groups {
+        let file = File::create(&path)?;
+        let mut writer = LineWriter::new(file);
+
+        if let Some(args) = format_lint_set(lints, &group) {
+            for arg in args {
+                writeln!(&mut writer, "{arg}")?;
+            }
+        };
+
+        writer.flush()?;
+    }
+
+    Ok(())
+}
+
+#[derive(Debug)]
+struct Args {
+    manifest_toml: PathBuf,
+    workspace_toml: Option<PathBuf>,
+    command: Command,
+}
+
+impl TryFrom<std::env::Args> for Args {
+    type Error = Cow<'static, str>;
+
+    fn try_from(mut args: std::env::Args) -> Result<Self, Self::Error> {
+        let _binary_path = args
+            .next()
+            .ok_or_else::<Cow<'static, str>, _>(|| "provided 0 arguments?".into())?;
+
+        let mut args = args.peekable();
+
+        // We get at least 'manifest-toml', and optionally a 'workspace-toml'.
+        let manifest_raw_arg = args
+            .next()
+            .ok_or(Cow::Borrowed("expected at least one arg"))?;
+        let manifest_toml =
+            try_parse_named_arg(&manifest_raw_arg, "manifest_toml").map(PathBuf::from)?;
+        let workspace_toml = args
+            .peek()
+            .and_then(|arg| try_parse_named_arg(arg, "workspace_toml").ok())
+            .map(PathBuf::from);
+        // If we got a workspace_toml arg make sure to consume it.
+        if workspace_toml.is_some() {
+            args.next();
+        }
+
+        // Use the remaining arguments to parse our command.
+        let command = Command::try_from(RemainingArgs(args))?;
+
+        Ok(Args {
+            manifest_toml,
+            workspace_toml,
+            command,
+        })
+    }
+}
+
+/// Tries to parse the value from a named arg.
+fn try_parse_named_arg<'a>(arg: &'a str, name: &str) -> Result<&'a str, String> {
+    arg.strip_prefix(&format!("--{name}="))
+        .ok_or_else(|| format!("expected --{name}=<value>, found '{arg}'"))
+}
+
+/// Arguments that are remaining after parsing the path to the `Cargo.toml`.
+struct RemainingArgs(std::iter::Peekable<std::env::Args>);
+
+#[derive(Debug)]
+enum Command {
+    /// Expects 4 filesystem paths in this order:
+    ///
+    /// 1. output for rustc lints
+    /// 2. output for rust check-cfg lints
```

Comment:
We should probably remove this expected arg, enum variant, and such until we actually support it. WDYT?

Reply:
Makes sense to me! Removed all references to it in the `cargo_lints` rule in the latest commit.